packages/api/test/utils/checkAgainstSpec.ts

```diff
@@ -19,17 +19,69 @@ ajv.addKeyword({
 
 ajv.addFormat("hex", /^0x[a-fA-F0-9]+$/);
 
+/**
+ * A set of properties that will be ignored during tests execution.
+ * This allows for a black-list mechanism to have a test pass while some part of the spec is not yet implemented.
+ *
+ * Properties can be nested using dot notation, following JSONPath semantic.
+ *
+ * Example:
+ * - query
+ * - query.skip_randao_verification
+ */
+export type IgnoredProperty = {
+  /**
+   * Properties to ignore in the request schema
+   */
+  request?: string[];
+  /**
+   * Properties to ignore in the response schema
+   */
+  response?: string[];
+};
+
+/**
+ * Recursively remove a property from a schema
+ *
+ * @param schema Schema to remove a property from
+ * @param property JSONPath like property to remove from the schema
+ */
+function deleteNested(schema: JsonSchema | undefined, property: string): void {
+  const properties = schema?.properties;
+  if (property.includes(".")) {
+    // Extract first segment, keep the rest as dotted
+    const [key, rest] = property.split(".", 2);
```

Comment:
Passing `limit` to `split` will not keep the `rest` as dotted. It will `exclude` the remaining values. 

From docs:
> Any leftover text is not included in the array at all.


```js
> "alpha.beta.gamma".split(".", 2)
[ 'alpha', 'beta' ]
```

Reply:
Good catch, I guess in practice this does not happen as we only have one level of nesting. Maybe just remove recursively calling deleteNested?

or could do
```js
const [key, ...rest] = property.split(".");
deleteNested(properties?.[key], rest.join("."));
```